src/state.ts

```diff
@@ -23,6 +28,28 @@ const workloadsLruCacheOptions: LruCache.Options<string, string> = {
   updateAgeOnGet: false,
 };
 
+/**
+ * Type is identical to the one in src/supervisor. However, it is declared here
+ * to prevent a dependency of this module to outside users and avoid circular refs.
+ * Additionally, we may want to know how our cache state is affected whenever the type changes.
+ */
+type IKubeObjectMetadataWithoutPodSpec = {
```

Comment:
`IKubeObjectMetadataWithoutPodSpec` can be moved from `src/state.ts` and `src/supervisor/workload-reader.ts` into `src/supervisor/types.ts` to resolve the circular reference.

Reply:
There's no real circular dep per se, I just found it weird that state depends on the supervisor module, which depends on the state module 🤔